src/core/connectionHelpers/connectionDetailsProvider.js

```diff
@@ -1,17 +1,20 @@
 import { IllegalArgumentException } from "../exceptions";
 import { ConnectionInfoType } from "./baseConnectionHelper";
+import { SESSION_TYPES } from "../../constants";
 
 export default class ConnectionDetailsProvider {
 
-  constructor(participantToken, chatClient) {
+  constructor(participantToken, chatClient, sessionType, getConnectionToken=null) {
     this.chatClient = chatClient;
     this.participantToken = participantToken || null;
     this.connectionDetails = null;
     this.connectionToken = null;
     this.connectionTokenExpiry = null;
+    this.sessionType = sessionType;
+    this.getConnectionToken = getConnectionToken;
   }
 
-  getConnectionToken() {
+  getFetchedConnectionToken() {
```

Comment:
this breaks backward compatibility. Is this a concern?

Reply:
This isn't an API... No customer should be calling this method.